src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs

```diff
@@ -266,7 +269,15 @@ private void SetUserPreferencesMaybeAsync(LANGPREFERENCES3[] languagePreferences
             }
             else
             {
-                Task.Factory.StartNew(() => this.SetUserPreferencesMaybeAsync(languagePreferences), CancellationToken.None, TaskCreationOptions.None, ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.TaskScheduler);
+                Task.Factory.StartNew(
+                    async () =>
+                    {
+                        await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
here you didn't pass the yield flag.  why not?

Reply:
➡️ The code is executing within a `Task.Factory.StartNew`, so we know the operation starts on the thread pool and will necessarily yield on the switch to the main thread. This code predates the introduction of the `alwaysYield` form in my branch. I believe the current conversion would be:

```csharp
ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
{
  await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true);
  this.SetUserPreferencesMaybeAsync(languagePreferences);
});
```

Updates to the form seen here are covered indirectly under the VSTHRD110 umbrella.